Converted to reusable address KPT

bnaolu opened the pull request:
- Files added:
   - address.html
   - address.md
   - 'state-combo-box.html' in includes folder
- lots of conditional statements in address.html 🥲  RIP to my brain. Thankful for that one python class I took LOL

Any and all feedback welcomed!

pglevy commented on _layouts/address.html:
```diff
@@ -0,0 +1,81 @@
+            </p>
+            <h2 class="margin-top-6">Contact Details</h2>
+            {% for label in page.contact-labels %}
+              <label class="usa-label" for="{{label.input-name}}">
```
It's nice to keep spacing with template placeholders consistent. Our convention so far has been to include a space before and after for variables like `{{ page.title }}`.

pglevy commented on _layouts/address.html:
```diff
@@ -0,0 +1,81 @@
+                  <abbr title="optional"></abbr>
+                {% endif %}              
+              </label>
+              <input 
```
Let's try to avoid line breaks within elements.

pglevy commented on _layouts/address.html:
```diff
@@ -0,0 +1,81 @@
+            {% for label in page.contact-labels %}
+              <label class="usa-label" for="{{label.input-name}}">
+                {{ label.label }}
+                {% if label.required %}
```
Nice approach to handling the required attribute!

pglevy commented on _layouts/address.html:
```diff
@@ -0,0 +1,81 @@
+
+            <div class="usa-checkbox margin-top-2">
+              <input class="usa-checkbox__input" id="receive-email-updates" type="checkbox" name="receive-email-updates" value="receive-email-updates">
+              <label class="usa-checkbox__label" for="receive-email-updates">Yes, I would like to receive updates on my status</label>
```
I think this is fine for this iteration, but since you've come with a way to separate all the other content, it might be nice to figure out a solution for these one-off checkboxes.

pglevy replied:
👍

pglevy commented on _layouts/address.html:
```diff
@@ -0,0 +1,81 @@
+                name="{{label.input-name}}" 
+                {% if label.required %} required {% endif %}
+                >
+              {% elsif label.form-type == "state-combo-box" %}
```
This looks like a typo. But it seems to be working. 🤷‍♂️ Maybe double check.

pglevy commented on _layouts/address.html:
```diff
@@ -0,0 +1,81 @@
+                >
+              {% elsif label.form-type == "state-combo-box" %}
+                {% include state-combo-box.html %}
+              {% else %}
```
If there's nothing to do for "else," you can just leave it out.

pglevy commented on _layouts/address.html:
```diff
@@ -0,0 +1,81 @@
+            <div class="usa-checkbox margin-y-5">
+              <input class="usa-checkbox__input" id="preferred-address" type="checkbox" name="preferred-address" value="preferred-address">
+              <label class="usa-checkbox__label margin-top-3 margin-bottom-2" for="preferred-address">Make this my preferred address</label>
+              <button class="usa-button  usa-button--unstyled">Add new</button>
```
Let's avoid extra spaces with class properties.

pglevy commented on _layouts/address.html:
```diff
@@ -0,0 +1,81 @@
+              <button class="usa-button  usa-button--unstyled">Add new</button>
+            </div>
+            <button class="usa-button margin-bottom-2">Save address</button>
+            <button class="usa-button  usa-button--unstyled display-block">Cancel</button>
```
Extra space

pglevy reviewed (changes requested):
Nice reusability!

bnaolu commented:
For recent commits today:
- fixed before and after spacing for variables
- linebreak issue changed to inline 
   - assuming this was for the input element
- added checkbox reusability code
- removed unneeded "else" conditional
- fixed spacing in button classes
